package/network/utils/comgt/files/ncm.sh

```diff
@@ -126,18 +137,27 @@ proto_ncm_setup() {
 		json_select ..
 	}
 
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
+#	json_get_vars connect
+#	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+#		echo "Failed to connect"
+#		proto_notify_error "$interface" CONNECT_FAILED
+#		return 1
+#	}
 
 	echo "Connected, starting DHCP on $ifname"
 	
 	proto_init_update "$ifname" 1
 	proto_send_update "$interface"
 
+	son_load "$(cat /etc/gcom/ncm.json)"
+	son_select "$manufacturer"
+	son_get_vars connect
+	val COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
```

Comment:
untested code. i doubt you runtime tested this otherwise you would have noticed the missing "j" in "son"